blogposts/2021/postgres-version-update.md

```diff
@@ -0,0 +1,141 @@
+---
+title: "Postgres version update"
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2021-03-23T18:00+02:00
+tags: [blog]
+slug: postgres-version-update
+heroImage: /blog/postgres-version-update.jpg
+socialImage: https://about.sourcegraph.com/blog/postgres-version-update.jpg
+published: true
+description: "As of Sourcegraph 3.27, we are updating the minimum supported version of Postgres from 9.6 to 12."
+---
+
+As of Sourcegraph 3.27, we are updating the minimum supported version of Postgres from 9.6 to 12.
+
+**If you are using the Sourcegraph maintained Docker images**: Your instance is already using an appropriate Postgres version and no action is needed on your part.
+
+**If you are maintaining an external database**: If your Postgres version is older than Postgres 12, you will need to update your database instance prior to upgrading from Sourcegraph 3.26 to 3.27.
+
+## Why are we forcing an upgrade?
+
+Postgres 9.6 was released on 2016/09/29 and since then there have been a [flood of changes](https://www.postgresql.org/about/featurematrix/) including:
+
+- Covering indexes for B-trees and GiST
+- SQL/JSON path expressions
+- Inlined WITH Queries (Common Table Expressions)
+- Parallel bitmap heap scans
+- Parallel hash and merge joins
+
+Apart from keeping the upgrade lag small so that we are not missing out on performance improvements and usable features, we have a technical impetus for upgrading at this time: Postgres 9.6 contains a SQL standard compatibility exclusion, as noted in the documentation for [creating triggers](https://www.postgresql.org/docs/9.6/sql-createtrigger.html#SQL-CREATETRIGGER-COMPATIBILITY):
+
+> PostgreSQL does not allow the old and new tables to be referenced in statement-level triggers, i.e., the tables that contain all the old and/or new rows, which are referred to by the OLD TABLE and NEW TABLE clauses in the SQL standard.
+
+Postgres 10 removes this compatibility restriction.
```

Comment:
This is phrased as a confusing double negative. Would be clearer to say

"Postgres 10 supports X"